websocket/websocket-server_test.go

```diff
@@ -0,0 +1,164 @@
+package websocket
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/websocket"
+
+	runtime "github.com/Hammad887/chat-app"
+)
+
+func TestRunWebsocketServer(t *testing.T) {
+	rt := &runtime.Runtime{} // Replace with your runtime initialization
+
+	// Create a test server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		RunWebsocketServer(rt)
+	}))
+	defer server.Close()
+	// Create an HTTP client for making requests to the test server
+	client := &http.Client{}
+	// Make an HTTP GET request to the WebSocket server's "/ws" endpoint
+	resp, err := client.Get(server.URL + "/ws")
+	if err != nil {
+		t.Fatalf("Failed to make an HTTP request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	// Assert that the response status code is as expected
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("Expected status code %d, but got %d", http.StatusOK, resp.StatusCode)
+	}
+}
+
+func TestClientSocket(t *testing.T) {
+	rt := &runtime.Runtime{} // Replace wi
+
+	// Create a test server for WebSocket connections
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ClientSocket(w, r, rt)
+	}))
+	defer server.Close()
+
+	// Simulate a WebSocket connection with the test server
+	u := "ws" + strings.TrimPrefix(server.URL, "http") + "/ws"
+	conn, _, err := websocket.DefaultDialer.Dial(u, nil)
+	if err != nil {
+		t.Fatalf("Failed to connect to WebSocket server: %v", err)
+	}
+	defer conn.Close()
+
+	type args struct {
+		responseBytes []byte
+		messageType   int
+		Conn          *websocket.Conn
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "CheckConnection",
+			args: args{
+				responseBytes: []byte(`{"requestType":"connect_socket"}`),
+				messageType:   websocket.TextMessage,
+				Conn:          conn, // Use the WebSocket connection to the server
+			},
+		},
+		{
+			name: "InvalidCheckConnection",
+			args: args{
+				responseBytes: []byte(`{"requestType": "connect_socket"}`),
+				messageType:   websocket.TextMessage,
+				Conn:          conn, // Use the WebSocket connection to the server
+			},
+		},
```

Comment:
You don't need to change this, but I have a suggestion for your work. This test doesn't cover the entire server, and there's a significant lack of coverage for each case in your server

Reply:
Will keep it under consideration in the future. Thank You!